### PR TITLE
CVSL-1280 add endpoint to get full size image for exclusion zone based on condition id

### PR DESCRIPTION
**Pull request by gurps-singh**
This PR is to add a new endpoint to the CVL public API to allow a JPEG image of an exclusion zone to be returned based on the ID of the condition that is required.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/PublicLicenceController.kt [hunk 1]
```diff
@@ -131,4 +131,46 @@ class PublicLicenceController(private val publicLicenceService: PublicLicenceSer
...
   )
   fun getLicenceByCrn(@PathVariable("crn") crn: String) = publicLicenceService.getAllLicencesByCrn(crn)
+
+  @GetMapping(
+    value = ["/licences/{licenceId}/conditions/{conditionId}/image-upload"],
+    produces = [MediaType.IMAGE_JPEG_VALUE],
+  )
+  @ResponseBody
+  @Operation(
+    summary = "Get the exclusion zone map image for a specified licence and condition",
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/PublicLicenceController.kt:141]: Get's an associated image upload for a specific licence and condition?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/PublicLicenceController.kt [hunk 2]
```diff
@@ -131,4 +131,46 @@ class PublicLicenceController(private val publicLicenceService: PublicLicenceSer
...
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "No image was found.",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getExclusionZoneImageByConditionId(
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/PublicLicenceController.kt:170]: getImageUpload

Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/publicApi/PublicLicenceController.kt:170]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/publicApi/PublicLicenceService.kt [hunk 3]
```diff
@@ -24,4 +29,26 @@ class PublicLicenceService(
 
     return licences.map { it.transformToPublicLicenceSummary() }
   }
+
+  @Transactional
+  fun getExclusionZoneImageByConditionId(licenceId: Long, conditionId: Long): ByteArray? {
```
[the remark this answers is not in the document]
Comment (gurps-singh) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/publicApi/PublicLicenceService.kt:34]: Done

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**